ArjunaJTS/jts/classes/com/arjuna/ats/internal/jts/ControlWrapper.java

```diff
@@ -270,7 +270,7 @@ public final void commit (boolean report_heuristics) throws Unavailable,
 		    // check termination status
 		    
 		    if (_controlImpl.getFinalStatus() != org.omg.CosTransactions.Status.StatusCommitted)
-                        throw new TRANSACTION_ROLLEDBACK();
+                        return;
```

Comment:
Are you sure about this one? If we tried to commit but ended up rolling back then we need the TRANSACTION_ROLLEDBACK exception.